python/ray/tests/test_scheduling.py

```diff
@@ -519,6 +519,26 @@ def f(s, ref):
     wait_for_condition(lambda: not fetches_queued())
 
 
+def test_nested_tasks(shutdown_only):
+    ray.init(num_cpus=1, resources={"worker": 3})
+
+    @ray.remote(num_cpus=1, resources={"worker": 1})
+    def h():
+        return 0
+
+    @ray.remote(num_cpus=1, resources={"worker": 1})
+    def g():
+        return ray.get(h.remote())
+
+    @ray.remote(num_cpus=1, resources={"worker": 1})
+    def f():
+        return ray.get(g.remote())
+
+    ready, _ = ray.wait(
+        [f.remote() for _ in range(1000)], timeout=60.0, num_returns=1000)
+    assert len(ready) == 1000, len(ready)
```

Comment:
Can we check the number of workers that were started?

Can we add a recursive test that checks for deadlock (and make sure it actually deadlocks if we don't add depth to the scheduling class)?

Reply:
Hmm the worker resource isn't released on `ray.get`, so I believe this test should be ensuring that only 3 workers are used. (I suppose it's possible that the worker pool is starting other idle workers, but that doesn't exactly seem relevant for this test?). 

I can add a comment about that, and add a test for the recursive case though.

Reply:
I see, that seems pretty dependent on the fact that we're not releasing the "worker" resource, though. Could we somehow add a check for the number of workers started or the number of concurrently running tasks (e.g., with a Counter actor)? It would be good to directly check the condition that we care about that.

Reply:
Bump?

Reply:
This uses the counter and not worker resource now?

I think this particular line of code just didn't change

Reply:
I see, I think I was looking at an old version of the PR.

I'm not sure where my comment went, but I think there is still an issue with this test because ray.wait doesn't throw exceptions. Can you use ray.get and make sure that this actually fails on master?